synapse/gene.py

```diff
@@ -162,10 +162,9 @@ def _eval(self, syms):
 class VarNode(GeneNode):
     def _eval(self, syms):
         name = self.tokn[1].get('name')
-        valu = syms.get(name)
-        if valu == None:
+        if name not in syms:
```

Comment:
it's a small nit, but this causes the ( potentially iterative in the case of a scope instance) resolving of that name multiple times for a single access.  in most cases, we've used a paradigm like:
```
# at top of module
class NoValu: pass
newp = NoValu()

# then some time later...
    valu = blah.get(name,newp)
    if valu is newp: # this checks id() equality, which requires same instance.
        raise NoSuchName(name=name)
    return valu
```

We should probably codify that with a class in synapse/common.py that we use for this purpose.  this saves us at least one ( and possibly several ) dictionary hash calculations and lookups.